File: Java/src/main/java/parrot/AfricanParrot.java
package parrot;

public class AfricanParrot extends Parrot {
    private int numberOfCocunuts;

    public AfricanParrot(ParrotTypeEnum type, int numberOfCoconuts, double voltage, boolean isNailed) {
        super(type, numberOfCoconuts, voltage, isNailed);
    }

    @Override
    public double getSpeed() {
        return super.getSpeed();
    }

    private double getLoadFactor() {
        return 9.0;
    }
}

File: Kotlin/src/main/kotlin/parrot/AfricanParrot.kt
package parrot

import kotlin.math.max

class AfricanParrot(
    private val numberOfCoconuts: Int
) : Parrot() {

    private val loadFactor: Double = 9.0

    private val minSpeed: Double = 0.0

    override val speed: Double
        get() = max(minSpeed, baseSpeed - loadFactor * numberOfCoconuts)
}
